pkg/handlers/generic/lifecycle/ccm/nutanix/handler_test.go

```diff
@@ -39,24 +37,61 @@ prismCentralAdditionalTrustBundle: "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVj
 
 # The Secret containing the credentials will be created by the handler.
 createSecret: false
-secretName: nutanix-ccm-credentials
-`
+secretName: nutanix-ccm-credentials`
 
 	expectedWithoutAdditionalTrustBundle = `---
 prismCentralEndPoint: prism-central.nutanix.com
 prismCentralPort: 9440
 prismCentralInsecure: true
-prismCentralAdditionalTrustBundle: ""
 
 # The Secret containing the credentials will be created by the handler.
 createSecret: false
-secretName: nutanix-ccm-credentials
-`
+secretName: nutanix-ccm-credentials`
+)
+
+var templateFile = filepath.Join(
+	moduleRootDir(),
+	"charts",
+	"cluster-api-runtime-extensions-nutanix",
+	"templates",
+	"ccm",
+	"nutanix",
+	"manifests",
+	"helm-addon-installation.yaml",
 )
 
 func Test_templateValues(t *testing.T) {
 	t.Parallel()
 
+	// Mimic the Helm templating using dummy values.
+	templateData := map[string]interface{}{}
+	require.NoError(
+		t,
+		unstructured.SetNestedField(
+			templateData,
+			true,
+			"Values",
+			"hooks",
+			"ccm",
+			"nutanix",
+			"helmAddonStrategy",
+			"defaultValueTemplateConfigMap",
+			"create",
+		),
+	)
+	var templatedBytes bytes.Buffer
+	require.NoError(
+		t,
+		template.Must(
+			template.New(
+				"helm-addon-installation.yaml").ParseFiles(templateFile),
+		).Execute(&templatedBytes, templateData),
+	)
+	cm := &corev1.ConfigMap{}
+	require.NoError(t, yaml.UnmarshalStrict(templatedBytes.Bytes(), cm))
+
+	valuesTemplate := cm.Data["values.yaml"]
```

Comment:
i honestly am very confused by whats going on here.

Reply:
Reading the configmap from the charts dir to get the actual template as would be rendered by Helm.

Reply:
oh ok i get it now. 

can you add more comments making this clear. alternatively you dont have to use the unstructured to set the helm bits and can extract the template some other way. 

definetly the changes in #819 would make things easier here :smile: